src/content/post/2024-05-06-rdf-shacl-and-kuzu.md

```diff
@@ -364,29 +373,42 @@ Both queries would return the same result:
+As can be seen, **you can choose the most appropriate query language** to analyze your data, depending on your
 workflow and how you want to interface with the graph -- using SPARQL via RDFLib or Cypher via Kùzu.
 Under the hood, Kùzu's query processor will use its native structured property
 graph model to plan and optimize the query, so there are no negative performance implications when using Cypher.
 
-**Note**: You can also extend Kuzu's RDFGraphs with other property graphs, and query both your triples
+You can also extend Kuzu's RDFGraphs with other property graphs, and query both your triples
 *and* the other property graphs with a uniform query language, Cypher. See Kùzu's [documentation](https://docs.kuzudb.com/rdf-graphs/rdfgraphs-overview#querying-of-regular-node-and-relationship-tables-and-rdfgraphs) page for more information.
 
+---
+
+### Note on performance
+When running SPARQL queries via RDFLib on top of a Kùzu backend, keep in mind that all the
+RDF triples are pulled into memory, so you would typically resort to
+querying the Kùzu RDF graph directly [using Cypher](https://docs.kuzudb.com/rdf-graphs/rdfgraphs-overview/)
+for larger graphs where the triples do not fit in memory.
+
+---
+
 ## Conclusions
 
 In this post, we showed how RDF data in Turtle format can be easily loaded into Kùzu using RDFLib. This was
 done by specifying Kùzu as a backend in the RDFLib plugin. We then demonstrated how SHACL shapes can be used to
-validate the RDF data, allowing users to create data graphs in RDF that satisfy a set of conditions.
-Kùzu provides a simple and intuitive interface to load, query and visualize RDF graphs, without compromising
+validate the RDF data via the pySHACL library, allowing users to create data graphs in RDF that satisfy a set of conditions.
+We also showed how Kùzu provides a simple and intuitive interface to load, query and visualize RDF graphs, without compromising
 scalability and performance, because the RDF triples are essentially mapped to Kùzu's native property graph model.
+Users    can decide whether to query the graph via SPARQL (via RDFLib) or via Cypher (directly in Kùzu).
 
-Taking this further, users can expand on the demonstrated workflow by creating more complex
-RDF graphs in their domains, define more intricate SHACL shapes, and apply more advanced functionality
-in RDFLib. For example, you can do reasoning over RDF graphs using the [OWL-RL](https://owl-rl.readthedocs.io/en/latest/owlrl.html) implementation available in RDFLib.
+Taking this further, you can expand on the demonstrated workflow by creating more complex
+RDF graphs in your domain, define more intricate SHACL shapes, and ask  more advanced questions on your data.
+For example, you can perform knowledge representation & reasoning tasks like transitive closure
+over your RDF graphs using the [OWL-RL](https://owl-rl.readthedocs.io/en/latest/owlrl.html) implementation
```

Comment:
Included the OWL-RL link here to point people to other libraries that can combine with RDFLib.